runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,221 @@
+import net.runelite.api.GameState;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.api.NPC;
+import net.runelite.api.NpcID;
+import net.runelite.api.Player;
+import net.runelite.api.Varbits;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Stealing Artefacts",
+	description = "Show the location and status of your Stealing Artefact task.",
+	tags = {"thieving", "stealing", "artefact", "artefacts", "captain", "khaled"},
+	enabledByDefault = false
+)
+
+public class StealingArtefactsPlugin extends Plugin
+{
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean inPortPiscariliusRegion;
+
+	private NPC captainKhaled;
+
+	@Override
+	protected void startUp()
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() -> stealingArtefactState = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		client.clearHintArrow();
+		stealingArtefactState = null;
+		captainKhaled = null;
+		overlayManager.remove(overlay);
+	}
+
+	private boolean hasArtefact()
+	{
+		ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (inventory != null)
+		{
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		StealingArtefactState state = StealingArtefactState.getState(client.getVar(Varbits.STEALING_ARTEFACT_STATE));
+
+		if (state != StealingArtefactState.DELIVER_ARTEFACT)
+		{
+			stealingArtefactState = state;
+			if (state == StealingArtefactState.NO_TASK || state == StealingArtefactState.FAILURE)
+			{
+				client.clearHintArrow();
+			}
+			else
+			{
+				client.setHintArrow(stealingArtefactState.getWorldPoint());
+			}
+		}
+		else if (captainKhaled != null)
+		{
+			client.setHintArrow(captainKhaled);
+		}
+		else
+		{
+			client.clearHintArrow();
```

Comment:
This should probably not clear the hint arrow unless it is more-sure the current arrow is its own. A lot of content has hint arrows.

Reply:
does that go for the entire method's hint arrow clearing or just the one in the else statement? For example, if the varbit changes to NO_TASK or FAILURE, I would think they the player is actively doing SA, as to get FAILURE the player would had have the artefact and were either caught or left the region, and to get NO_TASK they would need to speak to Khaled after a failed delivery.

Reply:
Anytime the plugin might clear the hint arrow, it should check to make sure the player is actually interacting with (or just stopped interacting with) the minigame.

Reply:
This does not appear to be resolved.